src/Emogrifier/CssInliner.php

```diff
@@ -781,7 +783,7 @@ private function splitCssAndMediaQuery($css)
      */
     private function copyInlinableCssToStyleAttribute(\DOMElement $node, array $cssRule)
     {
-        $newStyleDeclarations = $this->parseCssDeclarationsBlock($cssRule['declarationsBlock']);
+        $newStyleDeclarations = $this->parseCssDeclarationsBlock((string)$cssRule['declarationsBlock']);
```

Comment:
In what situation is `$cssRule['declarationsBlock']` not a `string`?  Maybe we need to look into that...

Reply:
It will always be a string, but the static code analysis of PhpStorm does not know this and hence needs some help …

Reply:
Is there no other way to tell PhpStorm, e.g. with a `/** @var string[] $cssRule */` preceding, or perpahs `/** @var string $declarationsBlock */ $declarationsBlock = cssRule['declarationsBlock']`?

I don't like the idea of having an unncessary cast in the actual PHP code in order to work around an issue with an IDE analysis tool.

Reply:
We could introduce a local variable for this and add a type annotation for it. Would that be a viable solution?

Reply:
Oh, I see that's actually the second solution you have proposed. I'll implement it in a second.

Reply:
> /** @var string[] $cssRule */

That first suggestion may be not so good, as they're probably not all strings - and from what I've read PhpStorm possibly doesn't support the `(string|int)[]` syntax.